services/motion/builtin/builtin.go

```diff
@@ -212,18 +212,24 @@ func (ms *builtIn) MoveOnGlobe(
 	heading float64,
 	movementSensorName resource.Name,
 	obstacles []*spatialmath.GeoObstacle,
-	linearVelocity float64,
-	angularVelocity float64,
+	motionCfg *motion.MotionConfiguration,
 	extra map[string]interface{},
 ) (bool, error) {
 	operation.CancelOtherWithLabel(ctx, builtinOpLabel)
 
 	kinematicsOptions := kinematicbase.NewKinematicBaseOptions()
-	kinematicsOptions.LinearVelocityMMPerSec = linearVelocity
-	kinematicsOptions.AngularVelocityDegsPerSec = angularVelocity
+
+	if motionCfg.LinearMPerSec != 0 {
+		kinematicsOptions.LinearVelocityMMPerSec = motionCfg.LinearMPerSec * 1000
+	}
+	if motionCfg.AngularDegsPerSec != 0 {
+		kinematicsOptions.AngularVelocityDegsPerSec = motionCfg.AngularDegsPerSec
+	}
+	if motionCfg.PlanDeviationM != 0 {
+		kinematicsOptions.PlanDeviationThresholdMM = motionCfg.PlanDeviationM * 1000
+	}
 	kinematicsOptions.GoalRadiusMM = 3000
```

Comment:
If `PlanDeviationM` is less than this value, should it be the minimum of that and this default?

Reply:
that makes sense, I will do
```
kinematicsOptions.GoalRadiusMM = math.Min(motionCfg.PlanDeviationM * 1000, 3000)
```